Tests/iaas/flavor-naming/flavor_names.py

```diff
@@ -234,6 +234,13 @@ def shorten(self):
         """return canonically shortened name as recommended in the standard"""
         if self.hype is None and self.hwvirt is None and self.cpubrand is None:
             return self
+        # For non-x86-64, don't strip out CPU brand for short name, as it contains the architecture
+        if self.cpubrand and self.cpubrand.cpuvendor != 'i' and self.cpubrand.cpuvendor != 'z':
+            cpubrand = self.cpubrand
+            # Delete non-relevant details for the short name, only keep brand
+            cpubrand.cpugen = 0
+            cpubrand.perf = ""
+            return Flavorname(cpuram=self.cpuram, disk=self.disk, cpubrand=cpubrand, gpu=self.gpu, ib=self.ib)
         return Flavorname(cpuram=self.cpuram, disk=self.disk, gpu=self.gpu, ib=self.ib)
```

Comment:
Don't mutate the existing cpubrand object.

```suggestion
        cpubrand = None
        if self.cpubrand and self.cpubrand.cpuvendor not in ('i', 'z'):
            # Delete non-relevant details for the short name, only keep brand
            cpubrand = CPUBrand()
            cpubrand.cpuvendor = self.cpubrand.cpuvendor
            cpubrand.cpugen = 0
            cpubrand.perf = ""
        return Flavorname(cpuram=self.cpuram, disk=self.disk, cpubrand=cpubrand, gpu=self.gpu, ib=self.ib)
```